scripts/workers/code_upload_submission_worker.py

```diff
@@ -469,48 +509,81 @@ def update_failed_jobs_and_send_logs(
-                    try:
-                        pod_log_response = (
-                            core_v1_api_instance.read_namespaced_pod_log(
-                                name=pod_name,
-                                namespace="default",
-                                _return_http_data_only=True,
-                                _preload_content=False,
-                                container=container.name,
+    try:
+        job_def = read_job(api_instance, job_name)
+        controller_uid = job_def.metadata.labels["controller-uid"]
+        pod_label_selector = "controller-uid=" + controller_uid
+        pods_list = core_v1_api_instance.list_namespaced_pod(
+            namespace="default",
+            label_selector=pod_label_selector,
+            timeout_seconds=10,
+        )
+        container_state_map = {}
+        for container in pods_list.items[0].status.container_statuses:
+            container_state_map[container.name] = container.state
+        for container_name, container_state in container_state_map.items():
+            if container_name in ["agent", "submission"]:
+                if container_state.terminated is not None:
+                    if container_state.terminated.reason == "Error":
+                        pod_name = pods_list.items[0].metadata.name
+                        try:
+                            pod_log_response = (
+                                core_v1_api_instance.read_namespaced_pod_log(
+                                    name=pod_name,
+                                    namespace="default",
+                                    _return_http_data_only=True,
+                                    _preload_content=False,
+                                    container=container_name,
+                                )
                             )
+                            pod_log = pod_log_response.data.decode("utf-8")
+                            cleanup_submission(
+                                api_instance,
+                                evalai,
+                                job_name,
+                                submission_pk,
+                                challenge_pk,
+                                phase_pk,
+                                pod_log,
+                                message,
+                            )
+                        except client.rest.ApiException as e:
+                            logger.exception(
+                                "Exception while reading Job logs {}".format(e)
+                            )
+                    elif (
+                        container_name == "submission"
+                        and container_state.terminated.reason == "Completed"
+                        and container_state_map.get("sidecar-container")
+                        and container_state_map.get(
+                            "sidecar-container"
+                        ).terminated
```

Comment:
@savish28 let's move this check to a separate function. This is not readable

Reply:
@Ram81 Updated